### PR TITLE
Tag multi-file-upload browser tests that fail on Azure, and skip those tests.

### LINKED ISSUE #9293: Run browser tests for azure through github test action
#### Is your feature request related to a problem? Please describe.

Allow for the ability to run azure tests in the tests workflow. We should probably figure out which file changes should trigger these tests, since it would add a lot of time to the tests, but we should make it possible, so all these tests don't have to be run locally each time.

https://github.com/civiform/civiform/pull/6705/files was the PR to disable these

#### Describe the solution you'd like.

Allow running azure tests in tests.yaml

### PR DESCRIPTION
**Pull request by lagosr-google**
### Description

Add tag to for multi-file-upload tests that fail on Azure. Skip those tests.

1 - Add tag: @multi-file-upload to failing tests
2 - Detect when tests are running against an Azure server
3 - Set a reverse grep list of tags to pass on to playwright to skip those tests.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [X] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [X] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [X] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [X] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [X] Created unit and/or browser tests which fail without the change (if possible)
- [X] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [X] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [X] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`

Fixes #9293

### CODE REVIEW

File: browser-test/src/applicant/application_review.test.ts
Comment (lagosr-google) [inline browser-test/src/applicant/application_review.test.ts:1]: The actual change is 1 line on #18 to add the tag.

All other changes occurred when running bin/fmt over the file.

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]
I'd like to continue this discussion on Slack before we move forward with blocking tests on specific cloud providers.

File: bin/run-browser-tests [hunk 1]
```diff
@@ -42,12 +42,20 @@ if [[ -z "$(docker ps --filter "name=${CONTAINER_NAME}" --format "{{.Names}}")"
     civiform-browser-test:latest
 fi
 
+# Detect if we are running tests with an Azure server, by looking for the Azurite
+# image in the docker container list.
+IS_AZURE_SERVER=
+if docker ps --filter "ancestor=mcr.microsoft.com/azure-storage/azurite" -q; then
+  IS_AZURE_SERVER=1
+fi
+
```
Comment (gwendolyngoetz) [inline bin/run-browser-tests:51]: (blocking) This won't work correctly if both the dev and browser test servers are running azure.

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]
Updating the `wait_for...sh` script like this isn't going to behave consistently. Any ad-hoc use of `--grep-invert` is going to override what is in the script.

The shell scripts already pass through arguments via the `$@` down to Playwright.

Keep the tag on the `application_review.test.ts` file, but the changes to the shell scripts are unneeded.

With the test tagged we just need to modify the [test workflow file](
https://github.com/civiform/civiform/blob/c93a43aa53cae315b3030de774f4aae79ac042cd/.github/workflows/tests.yaml#L144) to conditionally ignore the tag. Something like this 

```yaml
run: |
  if [[ "${{ matrix.cloud }}" == "azure" ]]; then
    bin/run-browser-tests-ci --shard="${{ matrix.shard_number }}/${{ strategy.job-total }}" --grep-invert "@skip-on-azure"
  else
    bin/run-browser-tests-ci --shard="${{ matrix.shard_number }}/${{ strategy.job-total }}"
  fi
```

Running it from the cli manually won't automatically pickup that you want to ignore it, but a bash alias can solve that 

```bash
alias run-browser-tests-azure='bin/run-browser-tests --grep-invert "@skip-on-azure"'
```

There should be a way to have the alias automatically created upon creation of the devcontainer.

File: browser-test/src/applicant/application_review.test.ts [hunk 2]
```diff
@@ -13,385 +13,387 @@ import {
...
-  test('all major steps with multiple file upload flag', async ({
-    page,
-    adminQuestions,
-    adminPrograms,
-    applicantQuestions,
-  }) => {
-    test.slow()
+  test(
+    'all major steps with multiple file upload flag',
+    {tag: ['@multi-file-upload']},
```
[the remark this answers is not in the document]
Comment (lagosr-google) [reply browser-test/src/applicant/application_review.test.ts]: done

File: bin/run-browser-tests (hunk 1 above)
Comment (lagosr-google) [reply bin/run-browser-tests]: removed this part.

**Comment by lagosr-google** [comment]
> Updating the `wait_for...sh` script like this isn't going to behave consistently. Any ad-hoc use of `--grep-invert` is going to override what is in the script.
> 
> The shell scripts already pass through arguments via the `$@` down to Playwright.
> 
> Keep the tag on the `application_review.test.ts` file, but the changes to the shell scripts are unneeded.
> 
> With the test tagged we just need to modify the [test workflow file](https://github.com/civiform/civiform/blob/c93a43aa53cae315b3030de774f4aae79ac042cd/.github/workflows/tests.yaml#L144) to conditionally ignore the tag. Something like this
> 
> ```yaml
> run: |
>   if [[ "${{ matrix.cloud }}" == "azure" ]]; then
>     bin/run-browser-tests-ci --shard="${{ matrix.shard_number }}/${{ strategy.job-total }}" --grep-invert "@skip-on-azure"
>   else
>     bin/run-browser-tests-ci --shard="${{ matrix.shard_number }}/${{ strategy.job-total }}"
>   fi
> ```
> 
> Running it from the cli manually won't automatically pickup that you want to ignore it, but a bash alias can solve that
> 
> ```shell
> alias run-browser-tests-azure='bin/run-browser-tests --grep-invert "@skip-on-azure"'
> ```
> 
> There should be a way to have the alias automatically created upon creation of the devcontainer.

Made the change to move the detection of the context and setting of the grep invert to the test definition file.

I think the spirit of the change is to skip this test on the CI run only.. so not making any changes to the manually cli runs by a dev is better imo ..   no need to add an alias imo.

**Review by gwendolyngoetz — APPROVED** [review]

**Comment by gwendolyngoetz** [comment]
If you need to skip this on the azure deploy as well just append `--grep-invert "@northstar|@skip-on-azure"` to [this line](https://github.com/civiform/civiform-staging-deploy/blob/982aeefd54d7395899e16a52cd3ca09724b05785/.github/workflows/azure_deploy.yaml#L86).

**Review by dkatzz — APPROVED** [review]
Right now, we're running azure tests only if the branch name contains azure. I'm assuming that after this change, we'll want to run azure always? https://github.com/civiform/civiform/blob/5c2115ace7b9c8a72ed33241842baced482961d7/.github/workflows/tests.yaml#L79-L83

**Comment by gwendolyngoetz** [comment]
@dkatzz if we're ready to enable it. I have something already I can push. I just need to rebase it.

**Comment by dkatzz** [comment]
> @dkatzz if we're ready to enable it. I have something already I can push. I just need to rebase it.

Thank you, @gwendolyngoetz !!

**Pull request merged**